core/kernel/user_access.c

```diff
@@ -37,7 +38,7 @@ TEE_Result copy_from_user(void *kaddr, const void *uaddr, size_t len)
 
 TEE_Result copy_to_user(void *uaddr, const void *kaddr, size_t len)
 {
-	uint32_t flags = TEE_MEMORY_ACCESS_WRITE | TEE_MEMORY_ACCESS_ANY_OWNER;
+	uint32_t flags = TEE_MEMORY_ACCESS_READ | TEE_MEMORY_ACCESS_ANY_OWNER;
```

Comment:
What if the target user address isn't writeable? Then well crash OP-TEE core, right?

Reply:
I reverted this change. This was necessary previously, but now no longer needed